azurerm/resource_arm_azure_firewall_network_rule_collection.go

```diff
@@ -0,0 +1,355 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-04-01/network"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAzureFirewallNetworkRuleCollection() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAzureFirewallNetworkRuleCollectionCreateUpdate,
+		Read:   resourceArmAzureFirewallNetworkRuleCollectionRead,
+		Update: resourceArmAzureFirewallNetworkRuleCollectionCreateUpdate,
+		Delete: resourceArmAzureFirewallNetworkRuleCollectionDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"azure_firewall_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"priority": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+
+			"action": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(network.AzureFirewallRCActionTypeAllow),
+					string(network.AzureFirewallRCActionTypeDeny),
+				}, true),
+				StateFunc:        ignoreCaseStateFunc,
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+
+			"rule": {
```

Comment:
🤔 would it make sense for Rules to be split out into their own sub-resource? Looking at the API Docs and [the SDK](https://github.com/Azure/azure-sdk-for-go/blob/v18.0.0/services/network/mgmt/2018-04-01/network/azurefirewalls.go#L47) I can't see any validation/limitation for this?